### ISSUE TITLE
RedisTimeoutException when Cluster loses a node

### ISSUE THREAD

**Issue by Efp95** [StackExchange/StackExchange.Redis#1091]
I'm looking for **High Availability**. My first approach was attempting to use Redis Sentinel with no success due to the package's capabilities.

I have a 6-noded cluster (3 masters & 3 slaves) in Docker. My app is able to connect to these 3 masters BUT when I turn off or pause a master container, my application throws a RedisTimeoutException because it is not able to reach the node it was connected before.

This is the output of running _cluster nodes_ command **before** pausing a master
![image](https://user-images.githubusercontent.com/12222605/54158279-26100a00-4418-11e9-9345-95c54474e1d0.png)

This is the output of running _cluster nodes_ command **after** pausing a master
![image](https://user-images.githubusercontent.com/12222605/54158411-74250d80-4418-11e9-96ad-6914cbefe9dc.png)

This is the exception my app throws
![image](https://user-images.githubusercontent.com/12222605/54158429-7d15df00-4418-11e9-82b8-b33b3a02da58.png)

Currently, I'm connecting to the 3 masters and 3 slaves
```
private const string Host = "10.0.0.2:7000,10.0.0.3:7001,10.0.0.4:7002,10.0.0.5:7003,10.0.0.6:7004,10.0.0.7:7005";

private static readonly ConnectionMultiplexer Connection = ConnectionMultiplexer.Connect(Host);
```

**Comment by Efp95** [comment]
@mgravell Sorry to tag you.
Could you please give me a hand on this?
I'm not sure if I'm doing it right.

**Comment by mgravell** [comment]
You're doing it right - I just haven't had time to investigate between other things, and I'm preparing to head to a conference. Sorry for delay, but ultimately all support here comes out of our own personal time.

Re the general scenario (not looking specifically at the report text) - losing a node is always going to be a little spikey, as reaching a firm "yeah, that node is toast" is tricky. You don't want to be too twitchy that you report any network blip or server blip as a node failure.

Apologies that I haven't been able to read the report in detail yet (I'm on mobile etc), but: does it recover? What kind of time factor are we talking here? Seconds? Minutes?

**Comment by Efp95** [comment]
So thankful you answered!
And YES, it recovers but after getting 3 timeouts exceptions (5 seconds per each)
To avoid data loss I'm setting a retry strategy. But not pretty sure how to deal with it.

On the other hand, I've experienced a different error which is described in #282

**Comment by mgravell** [comment]
The combined improvements in 2.0.588 and 2.0.571 should, we believe, address a wide range of timeout and performance problems. I'm closing all timeout issues as "believed fixed in 2.0.588"; if you still see problems, please report them as new issues so we can start investigation with fresh eyes.
